tests/unit/test_mps_gradient_detection.py

```
@@ -0,0 +1,206 @@
+def test_neuraldbg_detects_gradient_discrepancy():
+    """
+    Core test: NeuralDBG must detect when injected gradient differs
+    from the expected gradient.
+
+    This tests the EVENT CAPTURE system, not explain_failure().
+    We check that NeuralDBG's SemanticEvent log contains gradient-related events
+    with the right severity when we inject bad gradients.
+    """
+    from neuraldbg import NeuralDbg, EventType
+
+    grad_norm, grad_mean, correct_grad = get_cpu_ground_truth()
```

Ruff (0.15.15)

[warning] Unpacked variable `grad_norm` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)